### PR TITLE
PLAT-867 | TaxonomyTerms structure

### PR DESCRIPTION
**Pull request by kvas-damian**
* using https://github.com/lazychaser/laravel-nestedset for tree storage and manipulations

### CODE REVIEW

File: app/Http/Controllers/Api/PrivateApi/TaxonomyTermsApiController.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+use Illuminate\Http\Request;
+
+class TaxonomyTermsApiController extends ApiController {
+	public function __construct(Request $request) {
+		parent::__construct($request);
+		$this->resourceName = config('papi.resources.taxonomy-terms');
+	}
+
+	public function getByTaxonomy($taxonomyId) {
+		return $this->transformAndRespond(TaxonomyTerm::where('taxonomy_id', $taxonomyId)->get()->toFlatTree());
```
Comment (kvas-damian) [inline app/Http/Controllers/Api/PrivateApi/TaxonomyTermsApiController.php]: Currently, I return a flatTree (collection with `parentId` attribute) to the frontend and build the tree there. This allows us to use standard Fractal transformers and includes.
The alternative solution is to return a nested structure, which makes frontend easier, although I didn't manage to make Fractal work with nested resources correctly (there is a problem with collecting includes from all levels). Another solution would be to use a custom structure that bypasses Fractal Transformers, although at this moment I was afraid of making this endpoint exceptional.

What are your thoughts? Do you see any alternatives? Maybe you have some suggestions?

File: resources/assets/js/admin/components/taxonomies/TaxonomyTermItem.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<template>
+	<li>
+		{{ term.id }}. {{included.tags[term.tags[0]].name}}
+		<ul v-if="childTerms.length">
+			<wnl-taxonomy-term-item
+				v-for="childTerm in childTerms"
+				:key="childTerm.id"
+				:term="childTerm"
+				:included="included"
+				:terms="terms"
```
Comment (rogatty) [inline resources/assets/js/admin/components/taxonomies/TaxonomyTermItem.vue]: Consider using `left` and `right` to minimize the list size that we pass here.

Comment (kvas-damian) [reply resources/assets/js/admin/components/taxonomies/TaxonomyTermItem.vue]: We agreed offline that we'd like to apply performance optimizations after we migrate to vuex

Comment (rogatty) [reply resources/assets/js/admin/components/taxonomies/TaxonomyTermItem.vue]: Let's re-evaluate when we have the state in Vuex.

**Pull request merged**